database/sqlite/sqlite_health.c

```diff
@@ -0,0 +1,1357 @@
+            (long unsigned int)sqlite3_column_int64(res, 27),
+            (sqlite3_column_int64(res, 10) & HEALTH_ENTRY_FLAG_SILENCED)?"true":"false");
+
+        //not needed, it's already stored with replaced info
+        char *replaced_info = NULL;
+        if (likely(sqlite3_column_text(res, 20))) {
+            char *m = NULL;
+            replaced_info = strdupz((char *) sqlite3_column_text(res, 20));
+            size_t pos = 0;
+            while ((m = strstr(replaced_info + pos, "$family"))) {
+                char *buf = NULL;
+                pos = m - replaced_info;
+                buf = find_and_replace(replaced_info, "$family", (char *) sqlite3_column_text(res, 20) ? (const char *)sqlite3_column_text(res, 20) : "", m);
+                freez(replaced_info);
+                replaced_info = strdupz(buf);
+                freez(buf);
+            }
+        }
+
+        buffer_strcat(wb, "\t\t\"info\":\"");
+        buffer_strcat(wb, replaced_info);
+        buffer_strcat(wb, "\",\n");
+
+        if(unlikely(sqlite3_column_int64(res, 10) & HEALTH_ENTRY_FLAG_NO_CLEAR_NOTIFICATION)) {
+            buffer_strcat(wb, "\t\t\"no_clear_notification\": true,\n");
+        }
+
+        buffer_strcat(wb, "\t\t\"value\":");
+        if (sqlite3_column_type(res, 25) == SQLITE_NULL)
+            buffer_strcat(wb, "null");
+        else
+            buffer_rrd_value(wb, (calculated_number) sqlite3_column_double(res, 25));
+        buffer_strcat(wb, ",\n");
+
+        buffer_strcat(wb, "\t\t\"old_value\":");
+        if (sqlite3_column_type(res, 26) == SQLITE_NULL)
+            buffer_strcat(wb, "null");
+        else
+            buffer_rrd_value(wb, (calculated_number) sqlite3_column_double(res, 26));
+        buffer_strcat(wb, "\n");
+
+        buffer_strcat(wb, "\t}");
+
+        freez(replaced_info);
+    }
+
+    failed:
+    rc = sqlite3_finalize(res);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Failed to finalize statement for SQL_SELECT_HEALTH_LOG");
+}
+
+#define SQL_SELECT_ALL_HEALTH_LOG(guid,limit) "SELECT unique_id, alarm_id FROM health_log_%s order by unique_id desc LIMIT %u;", guid,limit
+void sql_health_alarm_log_select_all(BUFFER *wb, RRDHOST *host) {
+    sqlite3_stmt *res = NULL;
+    int rc;
+    char *guid = NULL, command[1000];
+
+    if (unlikely(!db_meta)) {
+        if (default_rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE)
+            error_report("Database has not been initialized");
+        return;
+    }
+
+    guid = strdupz(host->machine_guid);
+
+    guid[8]='_';
+    guid[13]='_';
+    guid[18]='_';
+    guid[23]='_';
+
+    sprintf(command, SQL_SELECT_ALL_HEALTH_LOG(guid, host->health_log.max));
+
+    rc = sqlite3_prepare_v2(db_meta, command, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement for SQL_SELECT_ALL_HEALTH_LOG");
+        return;
```

Comment:
guid memory leak

Reply:
Thanks, replaced with static variable as with other to keep similar.